### PR TITLE
Add myc proposal

### PR DESCRIPTION
**Pull request by raymogg**
# Motivation
Implementation for #1

### CODE REVIEW

File: proposals/12-myc-updated-proposal.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# Proposal Title
+Implementation for proposal 12, transferring 471k USDC to the Mycelium Multisig (0xa6a006C12338cdcDbC882c6ab97E4F9F82340651)
+
+## Implementation
+targets: [0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48]
+proposalData: [0x91ddeef8000000000000000000000000a6a006c12338cdcdbc882c6ab97e4f9f823406510000000000000000000000000000000000000000000000000000006da9c9a600]
```
Comment (sporejack) [inline proposals/12-myc-updated-proposal.md:6]: This section would greatly benefit from tabulation in my opinion - **block** code formatting at the very least (the wrapping really obfuscates this).

File: proposals/12-myc-updated-proposal.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+## Generation
+Generated using the following function call(s) and the DAOCheck tool
+```
```
Comment (sporejack) [inline proposals/12-myc-updated-proposal.md:10]:
```suggestion
```json
```

File: proposals/12-myc-updated-proposal.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+                {
+                    "name": "amount",
+                    "type": "uint256",
+                    "value": "471000000000"
+                }
+            ]
+        }
+    ]
+```
+Note that USDC uses 6 decimals and hence amount = 471000 * 10^6. The USDC address being used is 0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48
```
Comment (sporejack) [inline proposals/12-myc-updated-proposal.md:30]:
```suggestion
Note that USDC uses 6 decimals and hence `amount = 471000 * 10^6`. The USDC address being used is `0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48`.
```

File: proposals/12-myc-updated-proposal.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+# Proposal Title
+Implementation for proposal 12, transferring 471k USDC to the Mycelium Multisig (0xa6a006C12338cdcDbC882c6ab97E4F9F82340651)
```
Comment (sporejack) [inline proposals/12-myc-updated-proposal.md:2]:
```suggestion
Implementation for Proposal 12, transferring 471,000 USDC to the Mycelium multisig (`0xa6a006C12338cdcDbC882c6ab97E4F9F82340651`)
```

File: proposals/12-myc-updated-proposal.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
+# Proposal Title
```
Comment (sporejack) [inline proposals/12-myc-updated-proposal.md:1]: Likely just missed this but this should be a meaningful title.

**Review by sporejack — CHANGES_REQUESTED** [review]

**Comment by BenjaminPatch** [comment]
Do we think it would be beneficial to have the code used to generate the proposal data somewhere in this PR?

**Comment by raymogg** [comment]
> Do we think it would be beneficial to have the code used to generate the proposal data somewhere in this PR?

Yeah @BenjaminPatch for sure - that was the idea with including the JSON blob. I am working on open sourcing the DAOCheck tool today so I can basically reference that JSON and the DAOCheck tool as my method for generating the data.

If you don't use the tool I think people should put in code snippets of how they generated the data either using Ethers, Web3JS or some other library.

File: proposals/12-myc-updated-proposal.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+| :------------- | :-------------: |
+| targets | [0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48]
+| proposalData | [0x91ddeef8000000000000000000000000a6a006c12338cdcdbc882c6ab97e4f9f823406510000000000000000000000000000000000000000000000000000006da9c9a600]
+
+## Generation
+Generated using the following function call(s) and the DAOCheck tool
+```json
+"calls": [
+        {
+            "target": "USDC",
```
Comment (BenjaminPatch) [inline proposals/12-myc-updated-proposal.md:15]: Should we make the `target` field the address of USDC?

**Review by BenjaminPatch — CHANGES_REQUESTED** [review]

File: proposals/12-myc-updated-proposal.md (hunk 6 above)
Comment (raymogg) [reply proposals/12-myc-updated-proposal.md:15]: The DAOCheck tool uses aliases to autofill this (for ease of use) but in the proposal I've attached down the bottom the alias address used. which you can verify is the target in the proposal

**Review by sporejack — CHANGES_REQUESTED** [review]
Admittedly I missed this in my initial review, but this should link to a few places:
 - On-chain transaction (if relevant -- obviously not for Snapshot)
 - Off-chain voting artifact (if relevant -- very much so for Snapshot)

Otherwise this looks good.

**Comment by raymogg** [comment]
> Admittedly I missed this in my initial review, but this should link to a few places:
> 
> * On-chain transaction (if relevant -- obviously not for Snapshot)
> * Off-chain voting artifact (if relevant -- very much so for Snapshot)
> 
> Otherwise this looks good.

Good point. The issue does link the snapshot and discourse proposals, but I'll make that part of the proposal template too.

**Comment by sporejack** [comment]
> Good point. The issue does link the snapshot and discourse proposals, but I'll make that part of the proposal template too.

I saw the issue but I feel this document should describe the proposal in its entirety.

**Review by BenjaminPatch — APPROVED** [review]

**Review by sporejack — APPROVED** [review]

**Comment by raymogg** [comment]
Final confirmation of

- `0xa6a006C12338cdcDbC882c6ab97E4F9F82340651` is the correct Mycelium Multisig Address
- `0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48` is the USDC token address and is set as the target for the function call.
- USDC has 6 decimals and the hex value in proposalData of `0000000000000000000000000000000000000000000000000000006da9c9a600` is equal to`471000000000` which is `471000 * 10^6`
- The recipient is the Mycelium multisig address listed above
- The function signature in proposalData is `0xa9059cbb00000000000000000000000` which is the transfer function signature (see this example [USDC transfer](https://etherscan.io/tx/0x7d9aeeac09b3f0cc16067cfdbdf5f648d805a4ff812c436dc5bb311074ca36da))
- The ordering of the encoded parameters are `to` and `amount` as expected from transfer

**Review by BenjaminPatch — APPROVED** [review]
:fiji:

**Review by sporejack — APPROVED** [review]

**Pull request merged**